### DISCUSSION TITLE
Zustand subscribeWithSelector fires for all state changes

### DISCUSSION THREAD

**Discussion by pdhruv93** [pmndrs/zustand#1941] [Q&A] [answered]
I have subscribed to the accounts state of a QueryStore. But I can see "State changed" everytime even if the accounts state in not changed/updated.
```
let unsubscribe: any = null;

useEffect(() => {
    unsubscribe = useQueryStore.subscribe(
        (state) => [state.accounts],
        () => console.log('::::::State changed')
    );
    return () => unsubscribe?.();
}, []);
```

This is how I have created the store
```
const defaultProps: IQueryStoreProps = {
    accounts: [],
}
export const useQueryStore = create<IQueryStore>()(
    subscribeWithSelector((set, get) => ({
        ...defaultProps,
    }))
);
```

**Comment by dai-shi** [comment, marked as answer]
`(state) => [state.accounts],` creates a new array every time causing the changes. Can you do `(state) => state.accounts,` instead?

**Reply by pdhruv93** [reply]
@dai-shi yes it works this way but how can I specify multiple states to subscribe to within same statement?

**Reply by dai-shi** [reply]
In that case, you can specify `shallow` to the equalityFn option.

https://github.com/pmndrs/zustand/blob/f11cc7f37ad0a53f49370c0e4e659690a1c1a721/src/middleware/subscribeWithSelector.ts#L35
